testing/http_example_test.go

```diff
@@ -0,0 +1,100 @@
+package testing_test
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/cloudflare/tableflip"
+	"github.com/cloudflare/tableflip/testing"
+)
+
+type upgrader interface {
+	Listen(network, addr string) (net.Listener, error)
+	Stop()
+	Upgrade() error
+	Ready() error
+	Exit() <-chan struct{}
+}
+
+// This shows how to use the upgrader
+// with the graceful shutdown facilities of net/http
+// and using the stub implementation if on an unsupported platform.
+func Example_httpShutdown() {
+	var (
+		listenAddr = flag.String("listen", "localhost:8080", "`Address` to listen on")
+		pidFile    = flag.String("pid-file", "", "`Path` to pid file")
+	)
+
+	flag.Parse()
+	log.SetPrefix(fmt.Sprintf("%d ", os.Getpid()))
+
+	var upg upgrader
+	upgradeSupported := true
+	upg, err := tableflip.New(tableflip.Options{
+		PIDFile: *pidFile,
+	})
+	if errors.Is(err, tableflip.ErrNotSupported{}) {
+		upgradeSupported = false
+		upg, _ = testing.New()
+	} else if err != nil {
+		panic(err)
+	}
+	defer upg.Stop()
+
+	if upgradeSupported {
```

Comment:
Would the code fail if you skipped this check? It'd be nice if the only modification required would be to do `upg = testing.New()`.

Reply:
Technically it isn't required, since Windows cannot generate `SIGHUP`.
The reason I added it is because there's no reason to be running the useless goroutine.
So, no, it won't fail if you skip this check.
If you'd like I can remove the `if` and just put in a comment describing it for those who want to put it back.